components/AdminScoringDialog.tsx

```diff
@@ -0,0 +1,157 @@
+        </LabelText>
+
+        <LabelText label="Score" weight="regular">
+          <TextField.Root
+            id="gcseQualificationScore"
+            name="gcseQualificationScore"
+            type="number"
+            min={0.0}
+            max={10.0}
+            step={0.1}
+            className="flex-grow"
+            disabled={!gcseQualification}
+            required={!!gcseQualification}
+            defaultValue={parseFloat(row?.gcseQualificationScore?.toString() ?? '')}
+          />
+        </LabelText>
+      </Flex>
+
+      <Flex direction="column" gap="2">
+        <Heading as="h3" size="2">
+          Age 18 exam
+        </Heading>
+        <LabelText label="Type" weight="regular">
+          <Dropdown
+            values={Object.keys(AlevelQualification)}
+            currentValue={aLevelQualification}
+            onValueChange={setALevelQualification}
+            className="flex-grow"
+          />
+          <input name="aLevelQualification" type="hidden" value={aLevelQualification?.toString()} />
+        </LabelText>
+
+        <LabelText label="Score" weight="regular">
+          <TextField.Root
+            id="aLevelQualificationScore"
+            name="aLevelQualificationScore"
+            type="number"
+            min={0.0}
+            max={10.0}
+            step={0.1}
+            className="flex-grow"
+            disabled={!aLevelQualification}
+            required={!!aLevelQualification}
+            defaultValue={parseFloat(row?.aLevelQualificationScore?.toString() ?? '')}
+          />
+        </LabelText>
+      </Flex>
+
+      <LabelText label="Motivation Assessments">
+        <TextField.Root
+          id="motivationAdminScore"
+          name="motivationAdminScore"
+          type="number"
+          min={0.0}
+          max={10.0}
+          step={0.1}
+          defaultValue={parseFloat(internalReview?.motivationAdminScore?.toString() ?? '')}
+        />
+      </LabelText>
+
+      <LabelText label="Extracurricular Assessments">
+        <TextField.Root
+          id="extracurricularAdminScore"
+          name="extracurricularAdminScore"
+          type="number"
+          min={0.0}
+          max={10.0}
+          step={0.1}
+          defaultValue={parseFloat(internalReview?.extracurricularAdminScore?.toString() ?? '')}
+        />
+      </LabelText>
+
+      <LabelText label="Exam Comments">
+        <TextField.Root
+          id="examComments"
+          name="examComments"
+          defaultValue={internalReview?.examComments ?? undefined}
+        />
+      </LabelText>
+    </FormInDialog>
+  )
+}
+
+const AdminScoringDialog: FC<AdminScoringDialogProps> = ({ row }) => {
+  const formRenderer = ({ close }: { close: () => void }) => (
+    <AdminScoringForm close={close} row={row} />
+  )
+
+  return (
+    <GenericFormDialog title="Admin Scoring Form" form={formRenderer}>
+      <Button>Admin Scoring Form</Button>
+    </GenericFormDialog>
+  )
```

Comment:
(3) I would keep this component but:

- change `row` to `formData`
- extract at THIS level the form action i.e. `upsertAdminScoringWithId`
- compose the components as below:

```
<GenericDialog>
    <FormWrapper action={upsertAdminScoringWithId}>
        <AdminScoringForm data={data}/>
     </FormWrapper>
</GenericDialog>
```